exercises/07-normalizr/backend/server.ts

```diff
@@ -0,0 +1,77 @@
+import express from 'express'
+
+import { User, UserData } from '../src/modules/users/user-types'
+
+const port = 3001
+
+const areStringsSame = (a: string, b: string): boolean =>
+  a.toLowerCase() === b.toLowerCase()
+
+const hasSameName = (
+  user: UserData,
+  firstName: string,
+  lastName: string
+): boolean =>
+  areStringsSame(firstName, user.firstName) &&
+  areStringsSame(lastName, user.lastName)
+
+const findRegnalNumber = (users: User[], userName: UserData) => {
+  const regnalNumbers = users
+    .filter((user) => hasSameName(user, userName.firstName, userName.lastName))
+    .map(({ regnalNumber }) => regnalNumber)
+
+  if (regnalNumbers.length === 0) {
+    return 1
+  }
+
+  return Math.max(...regnalNumbers) + 1
+}
+
+const skills = [
+  {
+    id: 'skill-1',
+    name: 'Change my face',
+  },
+  {
+    id: 'skill-2',
+    name: 'Control dragons',
+  },
+]
+
+const users: User[] = []
+
+const app = express()
+app.use(express.urlencoded({ extended: true }))
+app.use(express.json())
+
+app.get('/users', (_req, res) => res.json(users))
+
+app.post('/users', (req, res) => {
+  const userName: UserData = req.body
+
+  const regnalNumber = findRegnalNumber(users, userName)
```

Comment:
I'd probably extract the bodies to a separate function

Reply:
Moved to separate functions, see 665b301b2eaede2ef38500b41dc2e5b90fef5e76 and  264464f